File: tests/core/resources/diff/pypi/class/diff.py




import random
import sys


class Game:
    def __init__(self, min=1, max=100):
        self.min = min
        self.max = max
        # Changed the name of this attribute
        self.value = random.randint(self.min, self.max)

    def play(self):
        print(f"I'm thinking of a number between {self.min} and {self.max}!")

        while True:
            try:
                guess = int(input("Enter your guess: "))
                
                if guess < self.value:
                    print("Too low!")
                elif guess > self.value:
                    print("Too high!")
                else:
                    print("You won!")
                    break
            except ValueError:
                print("Please enter a valid integer.")

File: tests/core/resources/diff/go/struct/diff.go


import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"strconv"
	"strings"
	"time"
)

type Game struct {
	min   int
	max   int
	value int
}

func NewGame(min, max int) *Game {
	rand.Seed(time.Now().UnixNano())
	return &Game{
		min:   min,
		max:   max,
		value: rand.Intn(max-min+1) + min,
	}
}

func (g *Game) Play() {
	fmt.Printf("I'm thinking of a number between %d and %d!\n", g.min, g.max)

	reader := bufio.NewReader(os.Stdin)

	for {
		fmt.Print("Enter your guess: ")
		input, _ := reader.ReadString('\n')
		input = strings.TrimSpace(input)
		guess, err := strconv.Atoi(input)

		if err != nil {
			fmt.Println("Please enter a valid integer.")
			continue
		}

		if guess < g.value {
			fmt.Println("Too low!")
		} else if guess > g.value {
			fmt.Println("Too high!")
		} else {
			fmt.Println("You won!")
			break
		}
	}
}
